src/modules/fancyzones/dll/dllmain.cpp

```diff
@@ -143,7 +152,29 @@ class FancyZonesModule : public PowertoyModuleIface
     {
         app_name = GET_RESOURCE_STRING(IDS_FANCYZONES);
         m_settings = MakeFancyZonesSettings(reinterpret_cast<HINSTANCE>(&__ImageBase), FancyZonesModule::get_name());
-        FancyZonesDataInstance().LoadFancyZonesData();
+        try
+        {
+            FancyZonesDataInstance().LoadFancyZonesData();
+        }
+        catch (...)
+        {
+            // Not being able to find path to persisted data completely breaks down FancyZones functionality.
+            // Retry again after 1 second, and if it fails again, disable core of FancyZones application.
+            std::this_thread::sleep_for(std::chrono::milliseconds(1000));
+            try
+            {
+                FancyZonesDataInstance().LoadFancyZonesData();
+            }
+            catch (...)
+            {
+                std::wstring errorMessage = GET_RESOURCE_STRING(IDS_FANCYZONES_DATA_ERROR) + L" " + NonLocalizable::PowerToysIssuesURL;
+                MessageBox(NULL,
+                           errorMessage.c_str(),
+                           GET_RESOURCE_STRING(IDS_POWERTOYS_FANCYZONES).c_str(),
+                           MB_OK | MB_ICONERROR);
+                disableFancyZonesModuleCore = true;

```

Comment:
call disable() here instead?

Reply:
Disable will do nothing here as `m_app` is null. This variable is important to make early exit in every subsequent call to `enable()`.

Reply:
After reviewing the code it seems like we can avoid introducing the `disableFancyZonesModuleCore` and just make sure that `m_app` is null. Calling `Disable(false)` will guarantee that `m_app` is set to null.

Reply:
That is true, but if runner calls `enable` again, we will enable powertoy, even though we shouldn't.

Reply:
If it's called again, we try again and in case we fail, not a big problem. I prefer to have less variable that effect the module state.

Reply:
First, FancyZones module is created by request from runner, and we fail to load FancyZones persisted data, and display that message box. Shortly after that, runner calls enable (since this message box was purely for user to see, FancyZones is created successfully from runner perspective), and we wan't to make sure that enable is blocked (that is why we have that early exit based on `disableFancyZonesModuleCore`).
Now, maybe it would be better that construction of FancyZones to return `nullptr` in the first place, which is fine, it's just that runner is completely lacking any sort of handling of anything that is not happy path.

Reply:
If we throw an exception here, the runner will catch it and the module will not be added to the `modules` list and therefore it will not call `enable()`

Reply:
Hm, than in fact all this processing here is not needed. I’ll remove it and check if it behaves as expected.

Reply:
Removed, tested.